### DISCUSSION TITLE
Add configuration to Assist Satellite entity

### DISCUSSION THREAD

**Discussion by synesthesiam** [home-assistant/architecture#1136] [Entity Models] [answered]
The Assist Satellite entity represents a remote voice satellite in Home Assistant. Satellites need a way to expose their current configuration to Home Assistant, and a way to update that configuration either from a user, a wizard, or an action.

We propose to extend the Assist Satellite entity with two abstract methods:

```python
@callback
def async_get_configuration(self) -> AssistSatelliteConfiguration:
    """Get the current satellite configuration."""

@abstractmethod
async def async_set_configuration(
    self, config: AssistSatelliteConfiguration
) -> None:
    """Set the current satellite configuration"""
```

These methods will be implemented by the platform, doing whatever is needed to get and set the configuration. `async_get_configuration` is a callback, indicating that the platform should retrieve and cache the current configuration for quick access.  `async_set_configuration` is an async method because the device should have completed setting the configuration by the time the method returns.

The `AssistSatelliteConfiguration` class may grow to cover many use cases, but for now we would like to focus on wake word detection. Specifically, enumerating the available on-device wake words and setting the active wake word(s).

```python
@dataclass
class AssistSatelliteWakeWord:
    """Available wake word model."""

    id: str
    """Unique id of wake word."""

    wake_word: str
    """Wake word phrase."""

    trained_languages: list[str]
    """List of languages that the wake word was trained on."""


@dataclass
class AssistSatelliteConfiguration:
    """Satellite configuration."""

    available_wake_words: list[AssistSatelliteWakeWord]
    """List of available available wake word models."""

    active_wake_words: list[str]
    """List of active wake word ids."""

    max_active_wake_words: int
    """Maximum number of simultaneous wake words allowed (0 for no limit)."""
```

Wake words have a unique identifier as well as their spoken phrase (e.g., "okay nabu").  The list of trained languages is available for separating language-specific variants of a given wake word.

The configuration itself contains the available wake words as well as the currently active ones. It's important to know the maximum number of active wake words because some devices have limited compute (e.g., ESP32-S3). A value of 0 for  `max_active_wake_words` indicates that the upper limit is not known (or is high enough not to matter practically).

## WebSocket Commands

Two websocket commands are proposed to make use of the new configuration methods:

`assist_satellite/get_wake_words` - gets a list of available wake words with the active ones marked

Example input:
```json
{ "entity_id": "assist_satellite.test" }
```
Example output: 
```json
{
  "max_active_wake_words": 1,
   "wake_words": [ 
      {  
        "id": "1234", 
        "wake_word": "okay nabu", 
        "trained_languages": ["en"], 
        "is_active": true
      }
    ]
}
```

`assist_satellite/set_wake_words` - sets the active wake words
 
Example input:
```json
{ 
  "entity_id": "assist_satellite.test" 
  "wake_words": ["1234"]
}
```

**Comment by balloob** [comment]
I wonder if `async_get_configuration` should be a callback? So we expect that to always be there. 

When a new pipeline comes in, we might need configuration from this, and time is of the essence. It would be up to the integration to ensure the config is always available.

**Reply by synesthesiam** [reply]
Updated the proposal to reflect this.

**Comment by synesthesiam** [comment]
Implementation: https://github.com/home-assistant/architecture/discussions/1136

**Comment by MartinHjelmare** [comment, marked as answer]
This architecture proposal has been accepted.
